src/wipplpy/modules/connection.py

```diff
@@ -0,0 +1,88 @@
+"""
+Define generic superclasses for accessing devices' databases.
+
+TODO: Add a MySQL connection and maybe other connection types as well?
+"""
+
+from abc import ABC, abstractmethod
+import socket
+
+import MDSplus as mds
+
+
+class MDSPlusConnection(ABC):
+    """
+    Create the generic superclass for generating MDSplus connections.
+    """
+    def __init__(self, ...):
+        self.is_remote_connection = None # bool tracking if conn. is remote
+
+    @abstractmethod
+    def make_connection(self, shot_number, tree_name, server_name):
```

Comment:
For an abstract method, there shouldn't be anything in it (i.e. it'll just be a def blah(...): pass). Maybe you should move the stuff in this method into a new private method? Then the instantiation of `make_connection` for the BRB and MST will just call this private method in their own ways.